### PR TITLE
Revised NFT Full Screen View (Buy Now + sub-tabs) #1344

### PR DESCRIPTION
**Pull request by Kamran-Khan-Gene**
Resolved reported issues.

## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

**Comment by kjawadDeveloper2** [comment]
@Kamran-Khan-Gene  pipelines are failing.

File: wallet/lib/model/nft.dart [hunk 1]
```diff
@@ -82,9 +82,7 @@ class NFT extends Equatable {
       final cookbook = await walletsStore.getCookbookById(cookbookID);
       ownerAddress = cookbook?.creator ?? "";
 
-      if (owner.isEmpty && type != NftType.TYPE_RECIPE) {
-        owner = await walletsStore.getAccountNameByAddress(ownerAddress);
-      }
+      owner = await walletsStore.getAccountNameByAddress(ownerAddress);
```
Comment (kjawadDeveloper2) [inline wallet/lib/model/nft.dart:85]: Whats the reason for removing?

File: wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart [hunk 2]
```diff
@@ -90,10 +91,11 @@ class _TabFieldState extends State<TabField> {
...
     final listDetails = nftDetail.entries
         .map(
-            (element) => _tabDetails(
-                field: element.key,
-                value: element.value,
-                customWidget: (element.key == LocaleKeys.recipe_id.tr() || element.key == LocaleKeys.ipfs_cid.tr()) && element.value.isNotEmpty ? _tabDetailsWithIcon(value: element.value) : null),
+          (element) => _tabDetails(
+            field: element.key,
+            value: element.value,
+            customWidget: (element.key == LocaleKeys.recipe_id.tr() || element.key == LocaleKeys.ipfs_cid.tr()) && element.value.isNotEmpty ? _tabDetailsWithIcon(value: element.value) : null,
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/widgets/tab_fields.dart:97]: Assign this value to a variable and then use it. ```(element.key == LocaleKeys.recipe_id.tr() || element.key == LocaleKeys.ipfs_cid.tr()) && element.value.isNotEmpty ? ```

File: wallet/lib/pages/home/currency_screen/model/ibc_coins.dart [hunk 3]
```diff
@@ -143,9 +143,18 @@ extension IBCCoinsDePar on IBCCoins {
...
+        }else if(denomAmount >= 100 && denomAmount < 1000){
+          return denomAmount.toStringAsFixed(0);
+        }else if(denomAmount >= 1000){
+          final RegExp reg = RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))');
+          return (denomAmount.round()).toString().replaceAllMapped(reg, (Match match) => '${match[1]},');
+        }
+        return amount;
       case IBCCoins.upylon:
-        return (double.parse(amount) / kBigIntBase).toStringAsFixed(0);
+        return (double.parse(amount) / kBigIntBase).toStringAsFixed(2);
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/home/currency_screen/model/ibc_coins.dart:157]: Pylons coin will never be in decimals.

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 4]
```diff
@@ -684,7 +669,32 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
               ),
             ),
           ),
-        )
+        ),
+        Align(
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/purchase_item/purchase_item_screen.dart:673]: Add a test for this.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

File: wallet/lib/model/nft.dart (hunk 1 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/model/nft.dart:85]: this condition restricting owner variable to get initialised, so the user is unable to view owner name in ownership tab in purchase nft screen. 


image

File: wallet/lib/pages/home/currency_screen/model/ibc_coins.dart [hunk 5]
```diff
@@ -143,9 +143,18 @@ extension IBCCoinsDePar on IBCCoins {
       case IBCCoins.ujuno:
       case IBCCoins.uatom:
       case IBCCoins.ustripeusd:
-        return (double.parse(amount) / kBigIntBase).toStringAsFixed(2);
+        final double denomAmount = double.parse(amount) / kBigIntBase;
+        if(denomAmount > 0 && denomAmount < 100){
```
[the remark this answers is not in the document]
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/home/currency_screen/model/ibc_coins.dart:147]: @StefanB90 wanted these formatting for USD dollar pricing, it is the implementation of these formatting.


image

File: wallet/lib/pages/home/currency_screen/model/ibc_coins.dart (hunk 3 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/home/currency_screen/model/ibc_coins.dart:157]: yes I did point out this thing on Friday's Scrum meeting, @StefanB90 it is intended and he want it to be like this.

File: wallet/test/unit_testing/pages/home/currency_screen/model/ibc_coins_test.dart [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:pylons_wallet/pages/home/currency_screen/model/ibc_coins.dart';
+import 'package:pylons_wallet/utils/constants.dart';
+
+void main(){
+
+  test('IBC Coins dollar conversion test', () {
```
[the remark this answers is not in the document]
Comment (Kamran-Khan-Gene) [reply wallet/test/unit_testing/pages/home/currency_screen/model/ibc_coins_test.dart:7]: this test case is checking formatting of pricing in USD. I will write another test for history tab.

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart (hunk 4 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/purchase_item/purchase_item_screen.dart:673]: Its test case is already added. I did not add this widget I changed its position in the widget tree.


image


image

File: wallet/lib/pages/home/currency_screen/model/ibc_coins.dart (hunk 3 above)
Comment (kjawadDeveloper2) [reply wallet/lib/pages/home/currency_screen/model/ibc_coins.dart:157]: @StefanB90 kindly provide info on this please.

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart (hunk 4 above)
Comment (kjawadDeveloper2) [reply wallet/lib/pages/purchase_item/purchase_item_screen.dart:674]: Then modify it so that it can also test its tap behavior.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

File: wallet/lib/pages/home/currency_screen/model/ibc_coins.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/home/currency_screen/model/ibc_coins.dart:147]: For 0.1 it will show 0.10.

**Comment by StefanB90** [comment]
please comment out #case uatom ujuno as we are only using ustripeusd & upylon .. perhaps to make it clear from a naming convention we call ustripeusd --> "ucash"  &  upylon --> "ucredit"? Not a requirement, but a suggestion if you think it would help?

If this issue relates to SKUs then we would need a separate issue to be raised to address it.

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

**Review by kjawadDeveloper2 — APPROVED** [review]

**Pull request closed without merging**